charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: pre-delete-job
+  namespace: nirmata
+  labels:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: hook-succeeded
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: delete-items
+        image: bitnami/kubectl:latest
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          #!/bin/sh
+          set -e
+
+          # Fetch the ConfigMap
+          configmap=$(kubectl get configmap items-created-configmap -n nirmata -o json)
+          
+          # Extract the items data
+          items=$(echo $configmap | jq -r '.data["items.json"]')
+
+          # Iterate through the items and delete them
+          echo $items | jq -c '.[]' | while read item; do
+            kind=$(echo $item | jq -r '.kind')
+            name=$(echo $item | jq -r '.name')
+
+            # Delete the item
+            kubectl delete $kind $name -n nirmata || true
```

Comment:
There are some cluster scoped resources like clusterroles and clusterrolebindings, no namespace there. Might work even if we give namespace, but need to test.

Reply:
it works even if you give a namespace, I gave that cause we dont know which one is clusterscoped and which isnt at this point